src/Appwrite/Platform/Workers/Messaging.php

```diff
@@ -339,21 +357,65 @@ private function email(Document $provider): ?EmailAdapter
     {
         $credentials = $provider->getAttribute('credentials');
         return match ($provider->getAttribute('provider')) {
+            'mock' => new Mock('username', 'password'),
             'mailgun' => new Mailgun($credentials['apiKey'], $credentials['domain'], $credentials['isEuRegion']),
-            'sendgrid' => new SendGrid($credentials['apiKey']),
+            'sendgrid' => new Sendgrid($credentials['apiKey']),
             default => null
         };
     }
 
     private function buildEmailMessage(Document $message, Document $provider): Email
     {
-        $from = $provider['options']['from'];
+        $fromName = $provider['options']['fromName'];
+        $fromEmail = $provider['options']['fromEmail'];
```

Comment:
Do we need to null check these?

Reply:
no, these are set up when creating provider.